packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -40,74 +51,87 @@ export function setResponsiveMode(responsiveMode: ResponsiveMode | undefined): v
 export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveMode }, TState>(
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>
 ): any {
-  const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
-    constructor(props: TProps) {
-      super(props);
-      this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
-
-      this.state = {
-        responsiveMode: _defaultMode || _lastMode || ResponsiveMode.large
-      };
-    }
+  // tslint:disable-next-line:function-name
+  function WithResponsiveModeBase(props: TProps, forwardedRef: React.Ref<React.Component<TProps, TState>>): JSX.Element | null {
```

Comment:
Is there a reason for using function syntax here instead of const and a lambda, like is typically done for function components?

Reply:
Isn't `function` syntax better? If you use the `const` form you have to declare the `I___Props` type twice, among other things.

Reply:
With our lint rules you shouldn't have to do it twice. I also prefer the const form since it makes it more explicit that the function is a component.
`const Foo: React.FunctionComponent<IFooProps> = props => { ... }`